pkg/cleanup/controller/controller.go

```diff
@@ -0,0 +1,251 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/sig-storage-local-static-provisioner/pkg/cleanup/deleter"
+	"sigs.k8s.io/sig-storage-local-static-provisioner/pkg/common"
+)
+
+type EntryPair struct {
```

Comment:
I don't think storing pointer to k8s objects in work queue is a good practice, because you need to frequently check if the pointer is nil and you may accidentally modify the obj itself accidentally. Considering splitting into two work queue for PV and Node, and only put name or the key of the object in the queue.

- pvQueue entryQueue workqueue.RateLimitingInterface
- nodeQueue entryQueue workqueue.RateLimitingInterface

Reply:
In that case, wouldn't it be easier to just have one workqueue with the PV names and then call the common.NodeAttachedToLocalPV(pv) that I created whenever I need the associated Node?

Reply:
Good idea! Having one work queue storing the pv names should be enough thanks to the function you created.

Reply:
Great thanks! Made this change